users/users/users.php

```diff
@@ -40,11 +43,97 @@ public function delete()
 		$this->plugin->setResponse( JText::_( 'PLG_API_USERS_IN_DELETE_FUNCTION_MESSAGE' ));
 	}
 
-	/**
-	 * Function post for create user record.
+	 /**
+	 * Function for edit user record.
 	 *
 	 * @return void
 	 */
+	public function put()
+	{
+		$db 			  = JFactory::getDbo();
+		$data             = array();
+		$eobj             = new stdClass();	
+		
+		// Get values paased by put in the url
+		$data = JRequest::get( 'put');
+		
+		// Check username or user_id to edit the details of user
+		if (!empty($data['username']) || !empty($data['user_id']))
+		{
+			if (empty($data['user_id']) && !empty($data['username']))
+			{
+				// Get user_id with the help of username
+				$query = $db->getQuery(true);
+				$query->select('id');
+				$query->from('#__users');
+				$query->where($db->quoteName('username') . '=' .  $db->quote($data['username']));
+				$db->setQuery($query);
+				$user_id = $db->loadResult();
+				$data['user_id'] = $user_id;
+			}
+			
+			// Given username or user_id not exist
+				if (!$data['user_id'])
+				{
+					$eobj->status = false;
+					$eobj->id = 0;
+					$eobj->code = '404';
+					$eobj->message = JText::_('COM_USERS_USER_NOT_FOUND');	
+					$this->plugin->setResponse($eobj);
+					return;
+				}
+				
+				$user = JFactory::getUser($data['user_id']);
+			
+				// Bind the data.
+				if (!$user->bind($data))
+				{
+					// User deatils are not updated
+					$message = $user->getError();
```

Comment:
Can you set the values of `$eobj->status` `$eobj->id` etc at the beginning of the script. And then change only those values that need to change. Will save you several lines of code.